src/op_traits.rs

```diff
@@ -1469,6 +1469,140 @@ impl Operator {
+        // `[Trap, ReadMem]` but not be a "normal" load. For example,
+        // `MemorySize` reads memory (or a property of it, anyway) but
+        // is not a "normal" load.
+        match self {
+            Operator::I32Load { .. }
+            | Operator::I64Load { .. }
+            | Operator::F32Load { .. }
+            | Operator::F64Load { .. }
+            | Operator::I32Load8S { .. }
+            | Operator::I32Load8U { .. }
+            | Operator::I32Load16S { .. }
+            | Operator::I32Load16U { .. }
+            | Operator::I64Load8S { .. }
+            | Operator::I64Load8U { .. }
+            | Operator::I64Load16S { .. }
+            | Operator::I64Load16U { .. }
+            | Operator::I64Load32S { .. }
+            | Operator::I64Load32U { .. }
+            | Operator::V128Load { .. }
+            | Operator::V128Load8x8S { .. }
+            | Operator::V128Load8x8U { .. }
+            | Operator::V128Load16x4S { .. }
+            | Operator::V128Load16x4U { .. }
+            | Operator::V128Load32x2S { .. }
+            | Operator::V128Load32x2U { .. }
+            | Operator::V128Load8Splat { .. }
+            | Operator::V128Load16Splat { .. }
+            | Operator::V128Load32Splat { .. }
+            | Operator::V128Load64Splat { .. }
+            | Operator::V128Load32Zero { .. }
+            | Operator::V128Load64Zero { .. }
+            | Operator::V128Load8Lane { .. }
+            | Operator::V128Load16Lane { .. }
+            | Operator::V128Load32Lane { .. }
+            | Operator::V128Load64Lane { .. } => true,
+            _ => false,
+        }
+    }
+
+    /// Is the operator an ordinary memory store?
+    ///
+    /// Note that this does not include all opcodes that write memory
+    /// *state*, such as growing memory, or that write and *also* read
+    /// memory, such as a memory copy; only "normal" store instructions.
+    pub fn is_store(&self) -> bool {
+        // We explicitly match on opcode rather than checking the
+        // effects set, beacuse some operators may have `[WriteMem]` or
+        // `[Trap, WriteMem]` but not be a "normal" load. For example,
+        // `MemoryFill` writes memory but is not a "normal" store.
+        match self {
+            Operator::I32Store { .. }
+            | Operator::I64Store { .. }
+            | Operator::F32Store { .. }
+            | Operator::F64Store { .. }
+            | Operator::I32Store8 { .. }
+            | Operator::I32Store16 { .. }
+            | Operator::I64Store8 { .. }
+            | Operator::I64Store16 { .. }
+            | Operator::I64Store32 { .. }
+            | Operator::V128Store { .. }
+            | Operator::V128Store8Lane { .. }
+            | Operator::V128Store16Lane { .. }
+            | Operator::V128Store32Lane { .. }
+            | Operator::V128Store64Lane { .. } => true,
+            _ => false,
+        }
+    }
+
+    /// Edit the `MemoryArg` if present.
```

Comment:
Wording suggestion: "Call `f` on the operator's `MemoryArg`, if it has one".

Reply:
Done!